catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -109,6 +137,34 @@ public void init() {
           res.header("Content-Encoding", "gzip");
         });
 
+    post(
+        "/cql/transform/:transformerId",
+        (req, res) -> {
+          String transformerId = req.params(":transformerId");
+          String body = util.safeGetBody(req);
+          CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+          CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
+
+          LOGGER.trace("Finding transformer to transform query response.");
+
+          ServiceReference<QueryResponseTransformer> queryResponseTransformer =
+              queryResponseTransformers
+                  .stream()
+                  .filter(transformer -> transformer.getProperty("id").equals(transformerId))
+                  .findFirst()
+                  .orElse(null);
+
+          if (queryResponseTransformer == null) {
+            LOGGER.debug("Could not find transformer with id: {}", transformerId);
+            res.status(404);
+            return mapper.toJson(ImmutableMap.of("message", "Service not found"));
+          }
+
+          attachFileWithTransformer(queryResponseTransformer, cqlQueryResponse, req, res);
+
+          return "";
```

Comment:
im getting comments to both add a variable yet not add a variable for this return statement. i think adding a small comment here would be a nice middle ground for that